src/main/java/org/cactoos/text/CapitalizeWordFunc.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import org.cactoos.Func;
+import org.cactoos.Text;
+
+/**
+ * Func that capitalize a word.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.3
+ */
+public final class CapitalizeWordFunc implements Func<Text, Text> {
+    @Override
+    public Text apply(final Text input) throws Exception {
+        return new FormattedText(
```

Comment:
@fabriciofx I'm flattered. Thanks.

you are right `text.asString().length()` in a Ctor is unacceptable. Should we just mimic the behaviour of the JDK that when we don't supply a endIndex, we just take everything util the end? 

If NOT.... I have an idea :-P
Instead of using SubText, we could introduce new HeadText(Text input, int endIndex), that gives the full text until the endIndex (Head of the Text), and a new TailText(Text input, int startIndex) which gives you everything from the startIndex until the end. Then there would be no confusion about how to signal you want subtext to return everything until the end.

```java
return new JoinedText(
  new UpperCaseText(
     new HeadText(input,0)),
  new TailText(input,1)
);
```

@yegor256 , WDYT about introducing Head/Tail-Text to avoid using "Flags" in SubText to signal intent to go until the end/start of the word?